reference_impl/reference_impl.rs

```diff
@@ -154,8 +154,7 @@ impl Output {
     }
 
     fn root_output_bytes(&self, out_slice: &mut [u8]) {
-        let mut output_block_counter = 0;
-        for out_block in out_slice.chunks_mut(2 * OUT_LEN) {
+        for (output_block_counter, out_block) in (0u64..).zip(out_slice.chunks_mut(2 * OUT_LEN)) {
```

Comment:
This is another place where I want to optimize for people being able to read this without necessarily knowing Rust. I do admit that I use `zip` in `words_from_little_endian_bytes` above, but that helper isn't really specific to BLAKE3, and hopefully the name makes it obvious what it does.